### DISCUSSION TITLE
IME options do not appear and explorer.exe crashes

### DISCUSSION THREAD

**Discussion by DG-7D** [valinet/ExplorerPatcher#588] [General]
This tool really helps me but I found a problem as in the title.

## Reproducing

1. Install ExplorePatcher and Japanese Microsoft IME
1. Set taskbar style to Windows 10
1. Change input language to Microsoft IME
1. Right-click on "A"
1. IME options do not appear
1. Within a minute, explorer.exe suddenly crashes and restarts

When ExplorerPatcher is not installed or taksbar style is set to Windows 11, IME options apeear correctly:
![imeoptions](https://user-images.githubusercontent.com/96519093/147179722-638f6ff5-e570-4f9c-9a67-08fd466de6f5.png)

## Environment

- Windows 11 build 22000.376
- ExplorerPatcher 22000.348.39.1

This caused on the clean VM too.

Thanks

**Comment by valinet** [comment]
Try on the latest pre release as well: right click taskbar - Properties - Updates - check Receive pre-release versions, if available, then click Update and install latest version.

**Reply by DG-7D** [reply]
Pre-release version does not work too.

**Reply by valinet** [reply]
Yeah, idk, has to be investigated. It’s hard doing Microsoft’s work without access to the same resources they have. Best way to have the Windows 10 interface remains staying on Windows 10. Maybe use these to enable full crash dumps and send that here or take a look on it, see if you can figure something out: https://docs.microsoft.com/en-us/windows/win32/wer/collecting-user-mode-dumps

**Comment by valinet** [comment]
UPDATE: Not required anymore, version 22000.376.40.19 and up include a fix for this issue (#598). The original post is left here for reference.

You can try using the previous version of Microsoft IME. The new one seems not to have been tested by them with the old taskbar, since that is not really supported in Windows 11. Idk, maybe a fix is found eventually, the crash is very hard to debug, it basically happens somewhere in the handler for some stuff that it does when showing the context menu. After a while, the callback returns with some unrecoverable error and it all crashes `explorer`. Pretty lame if you ask me, they cut a lot of corners with all the new stuff, everything is so abstracted and tightly coupled together and touted as such a big benefit, when in reality all the new stuff is crap that breaks when the slightiest variable changes, whereas old stuff works just fine 20 years after having been coded. Progress...

So, go to Settings - Time & language - Language & region, click the ... near "Japanase" and choose "Language options":

![image](https://user-images.githubusercontent.com/6503598/147396887-0fa1119a-a350-4163-811a-66f1559ed5dd.png)

Near "Microsoft IME", click the ... and choose "Keyboard options":

![image](https://user-images.githubusercontent.com/6503598/147396902-9a85e837-ceb5-414d-b924-5540ee2c2841.png)

In there, choose "General":

![image](https://user-images.githubusercontent.com/6503598/147396918-8dd9aefa-bbc6-42df-9676-15f738d83dcb.png)

Scroll all the way down and toggle "Use previous version of Microsoft IME" on:

![image](https://user-images.githubusercontent.com/6503598/147396932-caba51f7-9dde-4a62-b47c-1358d9ade2fb.png)

I don't know how crappy/better this is compared to what they ship in Windows 11, I personally have never used this, but it seems to at least offer a context menu:

![image](https://user-images.githubusercontent.com/6503598/147396953-eeea07c1-b40a-4b6f-b650-57faa01c6ba9.png)

If you ask me, I am sure it is some stupid skin or UWP stuff that they cannot access and instead of accounting for that, it's always easier to miserably fail because with new Windows, you either swallow all of it, including its many shitty parts, you either give up the entirety of it. Because #modularity and #progress.

Our world has become a sad joke...

**Reply by DG-7D** [reply]
Old version of IME is working, thanks.

I could find these difference only:
- Only old one has input mode (A/あ) notification (idk why this has been deleted)
- Only new one has dark mode

By the way, IME in Win10 looks between "old" and "new". This has no notification and no dark mode.... "previous version" option is there on Win10 too.

**Reply by valinet** [reply]
This should be fixed in the next pre-release (40.19). The context menu should now work with the new Microsoft IME as well. Track the progress of this and any new information about it here: #598
